geode-redis/src/main/java/org/apache/geode/redis/internal/ExecutionHandlerContext.java

```diff
@@ -196,22 +197,26 @@ public void channelInactive(ChannelHandlerContext ctx) {
 
   private void executeCommand(ChannelHandlerContext ctx, Command command) throws Exception {
     RedisResponse response;
-    if (isAuthenticated) {
-      if (command.isOfType(RedisCommandType.SHUTDOWN)) {
-        this.server.shutdown();
-        return;
-      }
 
-      response = command.execute(this);
+    if (command.isOfType(RedisCommandType.SHUTDOWN)) {
```

Comment:
Should this SHUTDOWN command code be moved to after we check of unsupported commands are allowed? I think SHUTDOWN is unsupported

Reply:
Good catch.